src/Persistence_matrix/include/gudhi/matrix.h

```diff
@@ -0,0 +1,2068 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hannah Schreiber
+ *
+ *    Copyright (C) 2022-24 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+/** @file matrix.h
+ * @author Hannah Schreiber
+ * @brief Contains @ref Gudhi::persistence_matrix::Matrix class.
+ */
+
+#ifndef MASTER_MATRIX_H
+#define MASTER_MATRIX_H
+
+#include <type_traits>
+#include <vector>
+#include <unordered_map>
+#include <map>
+#include <assert.h>
```

Comment:
In Gudhi we use GUDHI_CHECK so users can enable exceptions separately for gudhi or for their own code.
That mechanism is far from perfect (see #946, and I really don't like having to mention exceptions explicitly just to write an assertion), but that's what we currently have...